File: core/src/net/brenig/pixelescape/render/background/IBackgroundLayer.java
package net.brenig.pixelescape.render.background;

import net.brenig.pixelescape.render.WorldRenderer;

/**
 * Draws somthing in the game's background
 */
public interface IBackgroundLayer {

	void draw(WorldRenderer renderer);

	/**
	 * @return where a texture would need to be rendered to be {@code ratio} time slower than the foreground
	 */
	float getBackgroundWorldStart(WorldRenderer renderer, float ratio, int textureSize);
}

File: core/src/main/kotlin/net/brenig/pixelescape/render/background/IBackgroundLayer.kt
package net.brenig.pixelescape.render.background

import net.brenig.pixelescape.render.WorldRenderer

/**
 * Draws something in the game's background
 */
interface IBackgroundLayer {

    fun draw(renderer: WorldRenderer)

    /**
     * @return where a texture would need to be rendered to be `ratio` time slower than the foreground
     */
    fun getBackgroundWorldStart(renderer: WorldRenderer, ratio: Float, textureSize: Int): Float

    fun onResize(renderer: WorldRenderer)
}
